dds/DCPS/DataReaderImpl.cpp

```diff
@@ -3320,6 +3320,16 @@ DDS::ReturnCode_t DataReaderImpl::setup_deserialization()
     for (CORBA::ULong i = 0; i < repIds.length(); ++i) {
       Encoding::Kind encoding_kind;
       if (repr_to_encoding_kind(repIds[i], encoding_kind)) {
+          if (encoding_kind == Encoding::KIND_XCDR1) {
+            if (get_max_extensibility() == MUTABLE) {
```

Comment:
```suggestion
          if (encoding_kind == Encoding::KIND_XCDR1 && get_max_extensibility() == MUTABLE) {
```

Also the indentation seems messed up.